DToCToDB.md

```diff
@@ -0,0 +1,1193 @@
+
+dtoc.info()
+```
+
+    <class 'pandas.core.frame.DataFrame'>
+    Int64Index: 921880 entries, 0 to 15919
+    Data columns (total 16 columns):
+    Year                              921880 non-null int64
+    Month                             921880 non-null int64
+    Provider Parent Org Code          921880 non-null object
+    Provider Parent Name              921880 non-null object
+    Provider Org Code                 921880 non-null object
+    Provider Org Name                 921880 non-null object
+    Local Authority Code              921880 non-null object
+    Local Authority Name              921880 non-null object
+    Acute Or Non Acute Description    921880 non-null object
+    Reason For Delay                  921880 non-null object
+    NHS A SUM                         921843 non-null float64
+    NHS B SUM                         921825 non-null float64
+    Social Care A SUM                 921861 non-null float64
+    Social Care B SUM                 921848 non-null float64
+    Both A SUM                        921872 non-null float64
+    Both B SUM                        921874 non-null float64
+    dtypes: float64(6), int64(2), object(8)
+    memory usage: 119.6+ MB
+
+
+As noted above, we have a number of null values. There are two approches to dealing with this problem:
+* Drop the observation (row) entirely
+* Replace the null with a value
+
+In this case as we are dealing with numeric (float) values, therefore we are going to replace the null value with 0. So that other values in that observation are not lost. This is achieved using the `df.fillna()` function:
+
+
+```python
+dtoc = dtoc.fillna(0)
+```
+
+Finally we are going to add the datetime column. 
+
+We are going to make a pragmatic decision about the choice of date for each observation. If we look at the [definitions and guidance document](https://www.england.nhs.uk/statistics/wp-content/uploads/sites/2/2015/10/mnth-Sitreps-def-dtoc-v1.09.pdf) for this dataset; it shows that "NHS, Social Care and Both A SUM" are the number of patients classed as being in DToC on the **last Thursday of every month**. Whilst "NHS, Social Care and Both B SUM" are the total number of DToC days for all patients **during the entire month**.
+
+Therefore we are going to make a pragmatic decision to assign the day to the 1st of every month. Whilst this is technically incorrect, it won't impact on our future analysis purposes. Therefore we create a column called `Day`, which `=1` for all rows.
+
+We then combine the `Year`, `Month` and `Day` column into a temporary column called `Period`. We can then use `pd.to_datetime(dtoc['Period'])` assigned to a new column called `Date`.
+
+Finally we can drop `Year`, `Month`, `Day` and `Period` columns, as we no longer require these as they have been replaced by the `Date` column.
+
+
+```python
+# Day
+dtoc['Day'] = 1
+
+# Date
+dtoc['Period'] = dtoc['Year'].map(str) + '-' + dtoc['Month'].map(str) + '-' + dtoc['Day'].map(str)
```

Comment:
If you name the columns properly, you can just use [to_datetime](http://pandas.pydata.org/pandas-docs/stable/generated/pandas.to_datetime.html) without the string formatting.  I haven't tested this, but it should work:

```
dtoc["Date"] = pd.to_datetime(dtoc)
```

If it doesn't work, you may have to lowercase the column names `Month`, `Day`, and `Year`